src/main/webapp/homepage_script.js

```diff
@@ -113,21 +81,37 @@ if (JSON.stringify(neighborhood) != JSON.stringify([null, null])) {
     }
 }
 
-/* Function that display the help out confirmation overlay */
+/* Function that display the help out overlay */
 function helpOut(element) {
     const task = element.closest(".task");
-    const overlay = task.getElementsByClassName("confirm-overlay");
+    const overlay = task.getElementsByClassName("help-overlay");
     overlay[0].style.display = "block";
 }
 
-/* Function that hides the task after user confirms they want to help out */
-function removeTask(element) {
-    element.closest(".task").style.display = "none";
+/* Function sends a fetch request to the edit task servlet when the user
+offers to help out, edits the task's status and helper properties, and
+then reloads the task list */
+function confirmHelp(element) {
+    const task = element.closest(".task");
+    const url = "tasks/edit?task-id=" + task.dataset.key + "&action=helpout";
+    const request = new Request(url, {method: "POST"});
+    fetch(request).then((response) => {
+        // checks if another user has already claimed the task
+        if (response.status == 409) {
+            window.alert
+                ("We're sorry, but the task you're trying to help with has already been claimed by another user.");
+            window.location.href = '/';
+        }
+        // fetches tasks again if user's current neighborhood was successfully retrieved and stored
+        else if (JSON.stringify(neighborhood) != JSON.stringify([null, null])) {
```

Comment:
In addition to my other comment about this comparison, it might be worth creating a simple function called something like `userNeighborhoodIsKnown` that can do this comparison for you whenever you need it. That'll make it easier to update what it means for the neighborhood to be known, should that ever change.

Reply:
That's a good idea! I'll add a function to deal with this instead.